lib/exfmt/ast/to_algebra.ex

```diff
@@ -553,7 +553,7 @@ defmodule Exfmt.Ast.ToAlgebra do
     {call_args, [guard]} = Enum.split(args, -1)
     args_doc = args_to_algebra(call_args, ctx, opts)
     guard_doc = to_algebra(guard, new_ctx)
-    [args_doc, " when ", guard_doc]
+    [args_doc, group(nest(glue("", group(glue("when", guard_doc))), 1))]
```

Comment:
I don't think we ever want a space between the `when` and the guard to the right of it, so could we connect them using `space` instead of glue? :)